network/src/transport.rs

```diff
@@ -301,21 +319,33 @@ where
         chain_id: ChainId,
         application_protocols: SupportedProtocols,
     ) -> Self {
-        let mut own_handshake = HandshakeMsg::new(chain_id, network_context.network_id().clone());
-        own_handshake.add(SUPPORTED_MESSAGING_PROTOCOL, application_protocols);
+        // build supported protocols
+        let mut supported_protocols = BTreeMap::new();
+        supported_protocols.insert(SUPPORTED_MESSAGING_PROTOCOL, application_protocols);
```

Comment:
why aren't we just building the handshake once at the beginning of time and pass in clones later? or why are we only partially building it and then finishing it later? seems like we move an excessive amount of data ... motivation?

Reply:
that's what I point out in my PR description :) I think we can further optimize this by serializing it once and caching the serialized handshake message. But that's for a further PR as this is getting out of hand already.